### PR TITLE
feat: Added an extension to preview arbitrary URLs

### PR DESCRIPTION
**Pull request by ginqi7**
Added an extension to preview arbitrary URLs

add two command :
1. popweb-url-input
2. popweb-url-preview-pointer

file : current is only support relative path
image

http/https 
image

### CODE REVIEW

File: extension/url-preview/popweb-url.el [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  (let (
+        (match-result (match-string 0 current-line))
+        (current-path (when (buffer-file-name)
+                        (file-name-directory (buffer-file-name))
+                        ))
+        )
+        (replace-regexp-in-string "^file:" (concat "file:" current-path) match-result)
+        )
+  )
+
```
Comment (manateelazycat) [inline extension/url-preview/popweb-url.el]: Please clean those brackets

File: extension/url-preview/popweb-url.el [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+(defun popweb-current-line-url ()
+  "Return current line url."
+  (let (
+        (current-line (thing-at-point 'line t))
+        )
+    (when (string-match gnus-button-url-regexp current-line)
+      (popweb-url-local-file-url-completion current-line)
+      )
+    )
+  )
```
Comment (manateelazycat) [inline extension/url-preview/popweb-url.el]: Please clean those brackets

**Review by manateelazycat — CHANGES_REQUESTED** [review]

File: extension/url-preview/popweb-url.el (hunk 1 above)
Comment (ginqi7) [reply extension/url-preview/popweb-url.el]: update this.

**Comment by manateelazycat** [comment]
Looks good, thanks for patch. ;)

**Pull request merged**